src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -498,4 +498,40 @@ Tests:
     #    - { "int" : { "$numberLong" : "2000000000"}, "string" : "2000000000" }
     - "string": "2"
 
+  # Now moves with time so would always fail.
+  # - Name: Now
+  #   GivenTemplate:
+  #     date: {^Now: {}}
+  #   ThenReturns:
+  #   - { "date" : { "$date" : { "$numberLong" : "1453689676330" } } }
 
+  - Name: DateMinMax
+    GivenTemplate:
+      date: {^Date: {min: "2020-01-01", max: "2021-01-01"}}
+    ThenReturns:
+    - { "date" : { "$date" : { "$numberLong" : "1606971090942" } } }
```

Comment:
ermm..not sure how to read this or if indeed it's in the range lol..plz add a comment of what that date represents

Reply:
Yeah, Mongo BSON dates are -- hard to read.

Reply:
Mongo BSON date, I've added a comment to generate the time:

```
$> date --utc -d@$((1606971090942 / 1000))  +"%Y-%m-%dT%H:%M:%S.%3N"
2020-12-03T04:51:30.000
```


Making these more 'friendly' would be part of another ticket (see the last 2 comments). I'm not completely convinced of the merit of this part though.